pkg/endpoint/restore.go

```diff
@@ -222,13 +223,35 @@ func (e *Endpoint) restoreIdentity() error {
 	l, _ := labelsfilter.Filter(e.OpLabels.AllLabels())
 	e.runlock()
 
-	allocateCtx, cancel := context.WithTimeout(context.Background(), option.Config.KVstoreConnectivityTimeout)
-	defer cancel()
-	identity, _, err := e.allocator.AllocateIdentity(allocateCtx, l, true)
+	// Getting the ep's identity while we are restoring should block the
+	// restoring of the endpoint until we get its security identity ID.
+	// If the endpoint is removed, this controller will cancel the allocator
+	// requests.
+	controllerName := fmt.Sprintf("restoring-ep-identity (%v)", e.ID)
+	var (
+		identity *identity.Identity
+		gotIdty  = make(chan struct{})
```

Comment:
nit:

```suggestion
		allocatedIdentity  = make(chan struct{})
```

Reply:
(Mostly nitpicking on the `Idty` which is not that much shorter but presumably means `Identity` so we should be able to just use `Identity`)